core/src/main/java/org/elasticsearch/action/ingest/PutPipelineTransportAction.java

```diff
@@ -60,7 +72,28 @@ protected WritePipelineResponse newResponse() {
 
     @Override
     protected void masterOperation(PutPipelineRequest request, ClusterState state, ActionListener<WritePipelineResponse> listener) throws Exception {
-        pipelineStore.put(clusterService, request, listener);
+        NodesInfoRequest nodesInfoRequest = new NodesInfoRequest();
+        nodesInfoRequest.clear();
+        nodesInfoRequest.ingest(true);
```

Comment:
right, simulate api won't forward the request to an ingest node but always execute locally, forgot about that. Then this PR is good to go , unless we want to revise simulate.